packages/material-ui-styles/src/withStyles/withStyles.d.ts

```diff
@@ -16,20 +30,20 @@ export interface CSSProperties extends CSS.Properties<number | string> {
  *
  * if only `CSSProperties` are matched `Props` are inferred to `any`
  */
-export type StyleRules<Props extends object, ClassKey extends string = string> = Record<
+export type StyleRules<ClassKey extends string = string, Props extends object = {}> = Record<
```

Comment:
I guess core/styles has a different order and that is why we change it here? Breaking change in styles is better than core/styles but we have to make it somewhere.

Reply:
Now that I think about it: Can't we keep this inconsistency and import some "adapter type" from `core/styles`?

Reply:
It's a breaking change for core anyway. Regardless of the argument order because the number of arguments changed. [TypeScript doesn't support partial type inference](https://github.com/Microsoft/TypeScript/issues/26242).

Reply:
Correct

Reply:
If you've specified the argument then yes. If not then everything is fine.